src/box/memtx_tree.cc

```diff
@@ -886,7 +886,17 @@ memtx_tree_index_random(struct index *base, uint32_t rnd, struct tuple **result)
 	struct memtx_tree_index<USE_HINT> *index =
 		(struct memtx_tree_index<USE_HINT> *)base;
 	struct memtx_tree_data<USE_HINT> *res = memtx_tree_random(&index->tree, rnd);
-	*result = res != NULL ? res->tuple : NULL;
+	struct txn *txn = in_txn();
+	struct space *space = space_by_id(base->def->space_id);
+	if (res == NULL) {
+		*result = NULL;
+		memtx_tx_track_gap(txn, space, base, NULL, ITER_GE, NULL, 0);
+		return 0;
+	}
+	bool is_multikey = base->def->key_def->is_multikey;
+	uint32_t mk_index = is_multikey ? (uint32_t)res->hint : 0;
```

Comment:
We have special constant `MULTIKEY_NONE`.

Reply:
I don't understand how it is applicable here, please study this snippet:
https://github.com/tarantool/tarantool/blob/8ee0e434610189f3e4bbca0264de60aa79f0696e/src/box/memtx_tx.c#L2397-L2400

According to it, `memtx_tx_tuple_clarify` expects `0` — `MULTIKEY_NONE` is equal to `-1`.

Reply:
It's OK.